feat: add crate for pub modifier (#3271)
noir-lang/noir: docs/docs/language_concepts/01_functions.md

pub fn foo() {}
```

> You can however restrict the visibility of the function to its own crate, by specifying `pub(crate)`:

Reviewer: ```suggestion
You can also restrict the visibility of the function to only the crate it was defined in, by specifying `pub(crate)`:
```
